browser/src/control/Control.JSDialog.js

```diff
@@ -525,12 +532,36 @@ L.Control.JSDialog = L.Control.extend({
 		instance.posy = top + offsetY;
 
 		var width = instance.form.getBoundingClientRect().width;
-		var height = instance.form.getBoundingClientRect().height;
+		var autoFilterBottom = instance.posy + document.querySelector('.main-nav').clientHeight + document.getElementById('toolbar-wrapper').clientHeight + instance.form.getBoundingClientRect().height;
+		var toolbarTop = document.getElementById('spreadsheet-toolbar').getBoundingClientRect().top;
 		if (instance.posx + width > window.innerWidth)
 			instance.posx = window.innerWidth - width;
+
+		// at this point we have un updated potion of autofilter instance.
+		// so to handle overlapping case of autofiler and toolbar we need some complex calculation
+		if (autoFilterBottom > toolbarTop)
+			instance.posy = instance.posy - (autoFilterBottom - toolbarTop);
+
+		this.updatePosition(instance.container, instance.posx, instance.posy);
+	},
+
+	calculateSubmenuAutoFilterPosition: function(instance) {
+		// Get the position of the last focused element
+		var parentAutofilterEl = instance.lastFocusedElement.parentElement.getBoundingClientRect();
+		instance.posx = parentAutofilterEl.right + 20;
+		instance.posy = parentAutofilterEl.top - document.getElementById('toolbar-wrapper').offsetHeight;
+
+		var toolbarHeight= document.getElementById('spreadsheet-toolbar').offsetHeight;
```

Comment:
can we just rely on parent popup position?
to not depend on any other toolbar / element/ etc ?

Reply:
i have studied the JSON structure of child and parent popup they seem different but i don't find any specific property from JSON which can distingué that this JSON is for child popup.